fpdf/fpdf.py

```diff
@@ -751,6 +751,46 @@ def alias_nb_pages(self, alias="{nb}"):
         """
         self.str_alias_nb_pages = alias
 
+    def insert_page(
+        self, insert_location, orientation="", format="", same=False, duration=0, transition=None
+    ):
+        """
+        Inserts a new page in the document at a specified location.
+        A page will be added at the end of the PDF, and each page between
+        the specified index and the last page will me moved one page.
+
+        Args:
+            insert_location (int): The page number where the new page should be added.
+            orientation (str): "portrait" (can be abbreviated "P")
+                or "landscape" (can be abbreviated "L"). Default to "portrait".
+            format (str): "a3", "a4", "a5", "letter", "legal" or a tuple
+                (width, height). Default to "a4".
+            same (bool): indicates to use the same page format as the previous page.
+                Default to False.
+            duration (float): optional page’s display duration, i.e. the maximum length of time,
+                in seconds, that the page is displayed in presentation mode,
+                before the viewer application automatically advances to the next page.
+                Can be configured globally through the `.page_duration` FPDF property.
+                As of june 2021, onored by Adobe Acrobat reader, but ignored by Sumatra PDF reader.
+            transition (Transition child class): optional visual transition to use when moving
+                from another page to the given page during a presentation.
+                Can be configured globally through the `.page_transition` FPDF property.
+                As of june 2021, onored by Adobe Acrobat reader, but ignored by Sumatra PDF reader.
+        """
+        # Add a page at the end of the pdf
+        self.page = len(self.pages)
+        self.add_page(orientation, format, same, duration, transition)
```

Comment:
Unless `insert_location == self.pages_count`, the document last page should stay the same,
and hence its value for `orientation`, `format`, `duration` & `transition` should be taken from the last page before calling this method.

The same reasoning applies to all pages after `inert_location`: this method will break all those per-page settings by shfting them to previous page

Reply:
Line 781 ensures that we are at the last page.
```python
self.page = len(self.pages)
```